src/test/lib/DataTest.cpp

```diff
@@ -3556,6 +3553,263 @@ void QuicTestOperationPriority()
+
+                // NOTE: Splitting Start and Send without using QUIC_SEND_FLAG_START doesn't change operation order,
+                //       but it changes callback order. This is needed for using ExpectedXxxxOrder array.
+                if (i == ConnectionPriorityTestContext::NumSend-1) {
+                    Context.ExpectedStream = Streams[i];
+                    TEST_QUIC_SUCCEEDED(Streams[i]->Start(QUIC_STREAM_START_FLAG_PRIORITY_WORK))
+                    TEST_QUIC_SUCCEEDED(Streams[i]->Send(&Buffer, 1, QUIC_SEND_FLAG_FIN | QUIC_SEND_FLAG_PRIORITY_WORK));
+                } else {
+                    TEST_QUIC_SUCCEEDED(Streams[i]->Start(QUIC_STREAM_START_FLAG_NONE))
+                    TEST_QUIC_SUCCEEDED(Streams[i]->Send(&Buffer, 1, QUIC_SEND_FLAG_FIN));
+                }
+            }
+
+            TEST_QUIC_SUCCEEDED(Stream3.Start(QUIC_STREAM_START_FLAG_NONE));
+            TEST_QUIC_SUCCEEDED(Stream3.Send(&Buffer, 1, QUIC_SEND_FLAG_FIN));
+            Context.MaxSend = ConnectionPriorityTestContext::NumSend + 1;
+            Context.OperationQueuedComplete.Set(); // All operations are queued. Kick off processing the operations
+
+            MsQuicStream* ExpectedStartOrder[ConnectionPriorityTestContext::NumSend + 2] = {0};
+            ExpectedStartOrder[0] = &Stream1;
+            ExpectedStartOrder[1] = Streams[ConnectionPriorityTestContext::NumSend-1];
+            ExpectedStartOrder[ConnectionPriorityTestContext::NumSend+1] = &Stream3;
+            for (uint8_t i = 2; i < ConnectionPriorityTestContext::NumSend + 1; ++i) {
+                ExpectedStartOrder[i] = Streams[i-2];
+            }
+            MsQuicStream* ExpectedSendOrder[ConnectionPriorityTestContext::NumSend + 1] = {0};
+            ExpectedSendOrder[0] = Streams[ConnectionPriorityTestContext::NumSend-1];
+            ExpectedSendOrder[ConnectionPriorityTestContext::NumSend] = &Stream3;
+            for (uint8_t i = 1; i < ConnectionPriorityTestContext::NumSend; ++i) {
+                ExpectedSendOrder[i] = Streams[i-1];
+            }
+
+            TEST_TRUE(Context.AllReceivesComplete.WaitTimeout(TestWaitTimeout));
+
+            TEST_TRUE(memcmp(Context.StartOrder, ExpectedStartOrder, sizeof(ExpectedStartOrder)) == 0);
+            TEST_TRUE(memcmp(Context.SendOrder, ExpectedSendOrder, sizeof(ExpectedSendOrder)) == 0);
+            TEST_TRUE(Context.TestSucceeded);
+            for (uint8_t i = 0; i < ConnectionPriorityTestContext::NumSend; ++i) {
+                delete Streams[i];
+            }
+        }
+
+        // processing        | queued
+        // [1[s0]]           | []
+        // [1[s0]]           | [5[s0, s2]]
+        // [1[s0]]           | [4[s0p, s2p], 5[s0, s2]]
+        // [1[s0]]           | [4[s0p, s2p], 5[s0, s2]], 3[s0, s2]]
+        // [1[s0]]           | [4[s0p, s2p], 2[s0p, s2p], 5[s0, s2], 3[s0, s2]]
+        // [1[s0, s2]]       | [4[s0p, s2p], 2[s0p, s2p], 5[s0, s2], 3[s0, s2]]
```

Comment:
Could you check whether this scenario is valid? the explanation of this symbol is written around line 3639

Reply:
maybe correct